pyproject.toml

```diff
@@ -62,4 +62,133 @@ tag_prefix = "v"
 
 [tool.ruff]
 line-length = 88
+
+select = [
+    # pyflakes
+    "F",
+    # pycodestyle
+    "E",
+    "W",
+    # flake8-2020
+    "YTT",
+    # flake8-bugbear
+    "B",
+    # flake8-quotes
+    "Q",
+    # flake8-debugger
+    "T10",
+    # flake8-gettext
+    "INT",
+    # pylint
+    "PLC",
+    "PLE",
+    "PLR",
+    "PLW",
+    # misc lints
+    "PIE",
+    # flake8-pyi
+    "PYI",
+    # tidy imports
+    "TID",
+    # implicit string concatenation
+    "ISC",
+    # type-checking imports
+    "TCH",
+    # comprehensions
+    "C4",
+    # pygrep-hooks
+    "PGH",
+    # Ruff-specific rules
+    "RUF",
+    # pyupgrade
+    "UP",
+    # pydocstyle
+    "D",
+]
+
+target-version = "py38"
+ignore = [ # space before : (needed for how black formats slicing)
+    # "E203",  # not yet implemented
+    # do not assign a lambda expression, use a def
+    "E731",
+    # line break before binary operator
+    # "W503",  # not yet implemented
+    # line break after binary operator
+    # "W504",  # not yet implemented
+    # controversial
+    "B006",
+    # controversial
+    "B007",
+    # controversial
+    "B008",
```

Comment:
We might want to keep this consistent with geopandas, but I _think_ those could probably all be removed (they don't seem that controversial, and I also doubt we have occurrences of them)

Reply:
I agree that they sound like useful checks, and some af them can even avoid weird situations that are hard to debug.

Maybe activate them in geopandas as well, then it is consistent between geopandas and pyogrio?

Reply:
@theroggy to follow-up on this, I did clean up those settings on the geopandas side (https://github.com/geopandas/geopandas/pull/3291, and also adding isort support in https://github.com/geopandas/geopandas/pull/3297). So maybe we could just update here to simply follow the settings from geopandas now?

Reply:
@jorisvandenbossche Yes, I saw... I was busy looking that indeed (syncing settings with geopandas again), but real life got in the way ;-)... I'll try to finish it...

I added some additional things here as well, so it would be best these additions would be synced back to geopandas to get the settings completely in sync.